### PR TITLE
Regression test

### PR DESCRIPTION
**Pull request by jianyexi**
See regression result:
https://github.com/Azure/azure-openapi-validator/pull/224/checks?check_run_id=1152972393

### CODE REVIEW

File: regression/linting.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+export async function runLinter(readme: string) {
+  const linterCmd = genLintingCmd(readme)
+  try {
+    let linterErrors = await runCmd(linterCmd)
+    if (linterErrors.indexOf('{\n  "type": "') !== -1) {
+      linterErrors = cleanUpContent(linterErrors)
+      const errorJsonStr = "[" + linterErrors + "]"
+      const errorJson = JSON.parse(errorJsonStr).sort( (a,b) => {
+        let isLess = 0;
```
Comment (ruowan) [inline regression/linting.ts:18]: What does isLess do？

File: regression/linting.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+function cleanUpContent(s) {
+  let resultString = s.replace(/}\nProcessing batch task - {\"package-(.*).\n{/g, "},{")
+  resultString = resultString.replace(/{"package-(.*)} .\n/, "")
+  resultString = resultString.replace(/\nProcessing batch task(.*)./g, "")
+  resultString = resultString.replace(/}{/, "},{")
+  resultString = resultString.replace(/}\n{/, "},{")
+  return resultString
+}
+
+function reverse(s) {
```
Comment (ruowan) [inline regression/linting.ts:78]: comments reverse.

File: regression/linting.ts (hunk 1 above)
Comment (jianyexi) [reply regression/linting.ts:18]: It's the sort function's return value, may be -1,0,1 each indicate  < , = , >

File: regression/linting_all_readmes.test.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import { runLinter } from "./linting"
+import { allSpecs } from "./specsFilter"
+
+describe("validate all readme files", () => {
```
Comment (ruowan) [inline regression/linting_all_readmes.test.ts:4]: 9999 looks like magic number. Do you mean timeout?

File: .gitmodules [hunk 4]
```diff
@@ -0,0 +1,4 @@
+[submodule "regression/azure-rest-api-specs"]
```
Comment (ruowan) [inline .gitmodules:1]: Could you pls update `readme.md` about how to run regression test locally?

File: regression/linting_all_readmes.test.ts (hunk 3 above)
Comment (jianyexi) [reply regression/linting_all_readmes.test.ts:4]: right

File: .gitmodules (hunk 4 above)
Comment (jianyexi) [reply .gitmodules:1]: Thanks added

**Review by ruowan — APPROVED** [review]

**Pull request merged**